scipy/signal/spectral.py

```diff
@@ -511,22 +528,542 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+        raise ValueError('noverlap must be less than nperseg.')
+    noverlap = int(noverlap)
+
+    if isinstance(window, string_types) or type(window) is tuple:
+        win = get_window(window, nperseg)
+    else:
+        win = np.asarray(window)
+        if len(win.shape) != 1:
+            raise ValueError('window must be 1-D')
+        if win.shape[0] != nperseg:
+            raise ValueError('window must have length of nperseg')
+
+    step = nperseg - noverlap
+    binsums = np.sum((win[ii*step:(ii+1)*step] for ii in range(nperseg//step)),
+                     axis=0)
+
+    if nperseg % step != 0:
+        binsums[:nperseg % step] += win[-(nperseg % step):]
+
+    deviation = binsums - np.median(binsums)
+    return np.max(np.abs(deviation)) < tol
+
+
+def stft(x, fs=1.0, window='hann', nperseg=256, noverlap=None, nfft=None,
+         detrend=False, return_onesided=True, centered=True, padded=True,
+         axis=-1):
+    """
+    Compute the Short Time Fourier Transform (STFT).
+
+    STFTs can be used as a way of quantifying the change of a
+    nonstationary signal's frequency and phase content over time.
+
+    Parameters
+    ----------
+    x : array_like
+        Time series of measurement values
+    fs : float, optional
+        Sampling frequency of the ``x`` time series. Defaults to 1.0.
+    window : str or tuple or array_like, optional
+        Desired window to use. See `get_window` for a list of windows
+        and required parameters. If ``window`` is array_like it will be
+        used directly as the window and its length must be nperseg.
+        Defaults to a Hann window.
+    nperseg : int, optional
+        Length of each segment. Defaults to 256.
+    noverlap : int, optional
+        Number of points to overlap between segments. If ``None``,
+        ``noverlap = nperseg // 2``. Defaults to ``None``. When
+        specified, the COLA constraint must be met (see Notes below).
+    nfft : int, optional
+        Length of the FFT used, if a zero padded FFT is desired. If
+        ``None``, the FFT length is ``nperseg``. Defaults to ``None``.
+    detrend : str or function or ``False``, optional
+        Specifies how to detrend each segment. If ``detrend`` is a
+        string, it is passed as the ``type`` argument to the `detrend`
+        function. If it is a function, it takes a segment and returns a
+        detrended segment. If ``detrend`` is ``False``, no detrending is
+        done. Defaults to ``False``.
+    return_onesided : bool, optional
+        If ``True``, return a one-sided spectrum for real data. If
+        ``False`` return a two-sided spectrum. Note that for complex
+        data, a two-sided spectrum is always returned. Defaults to
+        ``True``.
+    centered : bool, optional
+        Specifies whether the input signal is padded via even extension,
+        to center the first window segment on the first input point.
+        I.e. ``[1, 2, 3, 4]`` is extended to ``[2, 1, 2, 3, 4, 3]``
```

Comment:
Actually `[1, 2, 3, 4]` would become `[-1,  0,  1,  2,  3,  4,  5,  6]`.  

I updated the docs to have a graphical example: https://github.com/scipy/scipy/pull/6699